src/battleSystem/hpGauge.mjs

```diff
@@ -3,110 +3,294 @@
+
+		// make the main bar
+		this.mainBar = new DynamicTwinBar (24, 6, this.hpShader, this.shader);
+
+		// initialisation
+		this.height = 10;
+		this.width = 100;
+
+		// make the reserves - a capacity setter could do this
+		// would make the constructor neater if we did that
+		let length = this.numReserves;
+		let slotXSize = this._slotXSize;
+		let slotYSize = this._slotYSize;
+		let slotX = this._width - slotXSize;
+		let slotY = this._height - slotYSize;
+		for (let i = 0; i < length; ++i)
+		{
+			this.reserves[i] = new Segment(slotX, slotY, slotXSize, slotYSize, this.hpShader, this.shader);
+			slotX -= (slotXSize - 1);
+		}
+	}
+
+	set hpColor (value)
+	{
+		if (this._hpColor !== value) {
+			this.tintColor.r = value.r;
+			this.tintColor.g = value.g;
+			this.tintColor.b = value.b;
+			this.tintColor.a = value.a * (1 - this._fadeness);
+			this._hpColor = value;
+			setVectorColor(this.backgroundShader, this.tintColor, 0.25);
+			setVectorColor(this.hpShader, this.tintColor, 1);
+		}
+	}
+
+	set x (value)
+	{
+		if (this._x !== value)
+		{
+			this._x = value;
+			
+			if ((this._numReservesFilled + this._numReservesDamaged) === this.numReserves)
+				this.mainBar.x = value + this._width - this.mainBar._width;
+			else
+				this.mainBar.x = value;
+
+			this.backgroundBar.x = value;
+			for (let i = 0, length = this.reserves.length; i < length; ++i)
+			{
+				this.reserves[i].parentX = value;
+			}
+		}
+	}
+
+	set y (value)
+	{
+		if (this._y !== value) {
+			this._y = value;
+			this.mainBar.y = value;
+			this.backgroundBar.y = value;
+			for (let i = 0, length = this.reserves.length; i < length; ++i)
+			{
+				this.reserves[i].parentY = value;
+			}
+		}
+	}
+
+	set slotYSize (value)
+	{
+		if (this._slotYSize !== value) {
+			this._slotYSize = value;
+			let slotY = this._height - value;
+			for (let i = 0, length = this.reserves.length; i < length; ++ i)
+			{
+				this.reserves[i].childY = slotY;
+			}
+		}
+	}
+
+	set slotXSize (value)
+	{
+		if (this._slotXSize !== value) {
+			this._slotXSize = value;
+			let slotX = this._width - value;
+			for (let i = 0, length = this.reserves.length; i < length; ++ i)
+			{
+				this.reserves[i].childX = slotX;
+				this.reserves[i].width = value;
+				slotX -= (value - 1);
+			}
+		}
+	}
+
+	//sort widths for top bar
+	set width (value)
+	{
+		if (this._width !== value) {
+			this._width = value;
+			this.slotXSize = this.maxSectors === 0
+				? this.slotXSize
+				: Math.ceil(this._width / (this.maxSectors - 1));
+					
+			if ((this._numReservesFilled + this._numReservesDamaged) === this.numReserves) {
+				this.mainBar.width = value * this.fraction;
+			}
+			else {
+				this.mainBar.width = value;
+			}
+			
+			this.backgroundBar.width = value;
+			for (let i = 0, length = this.reserves.length; i < length; ++i)
+			{
+				this.reserves[i].parentX = value + this._x;
+			}
```

Comment:
Hmm... my usual style is for one-liner loops to not have braces, but I guess that’s a potential maintenance issue moreso than `if` (you’re more likely to add stuff to a loop than a conditional).  So I guess we can keep the braces here, just please fix it so the `{` is on the same line as the `for` :)